plugins/content_distribution/lib/kContentDistributionFlowManager.php

```diff
@@ -1263,8 +1263,26 @@ public static function onMetadataChanged(Metadata $metadata, $previousVersion)
 			KalturaLog::info("Entry [".$entryId."] not found");
 			return true;
 		}
-		if($entry->getType() == entryType::LIVE_STREAM)
-			return true;
+	
+		$distributionProfiles = array();
+		$entryDistributions = EntryDistributionPeer::retrieveByEntryId($entryId);
+		$entryType = $entry->getType();
+		if($entryType == entryType::LIVE_STREAM)
```

Comment:
For better readability move this entire block to a function that should only decide if distribution is needed or not.